ynput/OpenPype
# Global: extract thumbnail with new settings

## Changelog Description
Settings are now configurable for the following:
- target size of thumbnail - source or constrained to specific 
- where should be frame taken from in sequence or video file
- if thumbnail should be integrated or not
- background color for letter boxes
- added AYON settings

## Additional info
![image](https://github.com/ynput/OpenPype/assets/40640033/59c41d9e-562d-409e-8288-b1b403191d13)


## Testing notes OpenPype:
1. It can be tested in TrayPublisher in `Render` family.
2. Try it on some sequence bigger then 1920x1080 (since we want to test that it was reformated to the format)
3. publish and if the thubmail is published check the size of it. 
4. you can enable `project_settings/global/publish/ExtractThumbnail/integrate_thumbnail` so the thumbnail is integrated and then check it in file browser under the published version representations.
5. change the size target `project_settings/global/publish/ExtractThumbnail/target_size/type` to `source` and check the size of published thubmail that is the same as the source resolution.


## Testing notes AYON:
1. make sure the latest core plugin version is deployed to your server (0.1.4)
2. notice there are new settings as screenshot shows 
![image](https://github.com/ynput/OpenPype/assets/40640033/8f02062c-fa4d-4937-bcb0-3c03d12e8d1f)
3. I am assuming you will be testing it on FullHD resolution sequence so best to test this is to resize to 1080x1080 as it is shown on picture
4. Follow Traypublisher Render product steps from above.
5. Thumbnail can be then observed from Loader at product version.
![image](https://github.com/ynput/OpenPype/assets/40640033/7bf6a412-408c-40df-93f2-57a880e760a8)



## Explicit Thumbnai workflow > development stages
[original PR](https://github.com/ynput/OpenPype/pull/5936)
This PR is just a placeholder and will be broken into smaller chunks:
- Colormanged Thumbnails - https://github.com/ynput/OpenPype/pull/5938	
- Ftrack integrate instance components refactory of the plugin for support of multiple thumbnail representations. https://github.com/ynput/OpenPype/pull/5939
- Nuke Explicit Thumbnail workfow https://github.com/ynput/OpenPype/pull/5941:
   - adding `need-thumbnail` tag and cross plugin flow
   - Nuke: removing extract_thumbnail 
   - improving flow of `publish_on_farm` tags
- adding new settings and implementation of new functions into extract thumbnail https://github.com/ynput/OpenPype/pull/5944:
	- integrating thubnails switch 	
	- size constrain 
	- make thumbnail from middle
